spec.bs

```diff
@@ -257,14 +257,191 @@ dictionary AuctionAdConfig {
 };
 </xmp>
 
+<div algorithm="runAdAuction()">
+
+The <dfn for=Navigator method>runAdAuction(|config|)</dfn> method steps are:
+
+1. If [=this=]'s [=relevant global object=]'s [=associated Document=] is not [=allowed to use=] the
+  [="run-ad-auction"=] [=policy-controlled feature=], then [=exception/throw=] a
+  "{{NotAllowedError}}" {{DOMException}}.
+1. Let |auctionConfig| be the result of running [=validate and convert auction ad config=] with
+  given |config| and [=validate and convert auction ad config/isTopLevel=] set to true. If it fails,
+  return [=a new promise=].

```

Comment:
I think it fails by throwing an exception, which will automatically convert to a promise, so you don't need that last sentence:
```suggestion
  given |config| and [=validate and convert auction ad config/isTopLevel=] set to true.
```